client/package.json

```diff
@@ -42,25 +35,16 @@
     "react-icons": "^5.0.1",
     "react-markdown": "^8.0.6",
     "react-papaparse": "^4.4.0",
-    "react-router-dom": "^6.4.5",
-    "react-scripts": "5.0.1",
+    "react-router-dom": "^6.22.3",
     "remark-gfm": "^3.0.1",
     "remirror": "^2.0.38",
-    "typescript": "5.1.6",
-    "unist-util-visit": "^4.1.2",
-    "web-vitals": "^2.1.4"
+    "unist-util-visit": "^4.1.2"
   },
   "scripts": {
-    "start": "react-scripts start",
-    "build": "react-scripts build",
-    "test": "react-scripts test",
-    "eject": "react-scripts eject"
-  },
-  "eslintConfig": {
-    "extends": [
-      "react-app",
-      "react-app/jest"
-    ]
+    "start": "vite",
+    "build": "tsc --noEmit && vite build",
+    "preview": "vite preview",
+    "test": "tsc --noEmit && vitest"
```

Comment:
Question: Would it make sense to add a Github Action for client tests (as well as `tsc`) alongside the API test workflow? (will make a separate PR for this if so)

Reply:
Yes! Though I think my preference personally is focusing on writing e2e tests with something like cypress